packages/vite/src/resolver.ts

```diff
@@ -29,7 +50,10 @@ export function resolver(): Plugin {
     },
     load(id) {
       if (id.startsWith(virtualPrefix)) {
-        return virtualContent(id.slice(virtualPrefix.length), resolverLoader.resolver);
+        let { src, watches } = virtualContent(id.slice(virtualPrefix.length), resolverLoader.resolver);
+        virtualDeps.set(id, watches);
```

Comment:
prefix here the src with `import.meta.glob(...watches)`. and filter watches for the ones containing globs. 
files should just work with addWatchFile.
like that you do not need the `configureServer` code

Reply:
Our actual use cases need more sophisticated filtering than we can do with import.meta.glob. For example, we need a way to include "node_modules/some-addon/_app_/components/foo.js" but only if "./components/foo.js" doesn't exist. If we do that with globs you will always evaluate the addon copy, even when it should not evaluate.

Reply:
mport.meta.glob is not eager, but can be configured to be so with params, so it shouldn't evaluate. but not sure if thats enough for the use case you are looking for

Reply:
Right, in the case of entrypoints we need the `eager: true` variant.